libtransmission/announcer.c

```diff
@@ -1249,9 +1249,11 @@ static void on_announce_done(tr_announce_response const* response, void* vdata)
                then a separate scrape isn't needed */
             if (scrape_fields >= 3 || (scrape_fields >= 1 && tracker->scrape_info == NULL))
             {
+                i = round(tier->scrapeIntervalSec * 0.95) + 
+                    tr_rand_int_weak(round(tier->scrapeIntervalSec * 0.1));
                 tr_logAddTorDbg(tier->tor, "Announce response contained scrape info; "
-                    "rescheduling next scrape to %d seconds from now.", tier->scrapeIntervalSec);
-                tier->scrapeAt = get_next_scrape_time(announcer->session, tier, tier->scrapeIntervalSec);
+                    "rescheduling next scrape to %d seconds from now.", i);
+                tier->scrapeAt = get_next_scrape_time(announcer->session, tier, i);
```

Comment:
Since this is done in two places, could you extract this into a static method, e.g. something like

```c
static int getRandomizedInterval(int seconds) {
   ...
}

...

i = getRandomizedInterval(tier->scrapeIntervalSec);
tr_logAddTorDbg(...

...

i = getRandomizedInterval(tier->announceIntervalSec);
tr_logAddTorDbg(...)
```

Reply:
We're already calling a static tr_rand_int_weak to get the random interval. The more accurate name for a new static function would be getSkewedInterval and make the arguments: tier->scrapeIntervalSec, and a maximum skew percentage. Someone else will need to make the PR for the change, documentation, and tests. I no longer have the time.

```
static int getSkewedInterval(int seconds, float percent) {
    return round(seconds * percent) + tr_rand_int_weak(round(seconds * percent));
}
```